src/apportioned-emissions/annual/annual-apportioned-emissions.service.ts

```diff
@@ -27,81 +29,87 @@ export class AnnualApportionedEmissionsService {
     req: Request,
     params: PaginatedAnnualApportionedEmissionsParamsDTO,
   ): Promise<AnnualUnitDataView[]> {
-    let entities: AnnualUnitDataView[];
-
-    try {
-      entities = await this.repository.getEmissions(
-        req,
-        fieldMappings.emissions.annual.data.aggregation.unit,
-        params,
+    return withSlaveConnection(this.dataSource, async () => {
+      let entities: AnnualUnitDataView[];
+
+      try {
+        entities = await this.repository.getEmissions(
```

Comment:
The `withSlaveConnection` function does target the replica database, but only if the **`EntityManager`** passed to the callback is used. This can be done by instantiating a new repository instance and passing the replica entity manager as an argument:

```suggestion
    return withSlaveConnection(this.dataSource, async (replicaManager: EntityManager) => {
      let entities: DayUnitDataView[];

      try {
        entities = await (new DayUnitDataRepository(replicaManager)).getEmissions(
```